pie/storage/__init__.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import annotations
+
+from pydoc import locate
+from typing import Any
+
+from nextcord.ext import commands
+
+from pie.storage.database import StorageData
+
+
+def get(module: commands.Cog, guild_id: int, key: str, default_value=None) -> Any:
+    """Get data from persistant DataStorage base on module and guild.
+    For saving global data (non-guild related) set guild_id to 0.
+
+    If value is not found in DB or it's type is unknown,
+    it returns default_value, which if not set is None.
+
+    Args:
+        module (:class:`nextcord.ext.commands.Cog`): module connected with the value
+        guild_id (:class:'int'): ID of guild connected with the value (0 for global)
+        key (:class:'str'): Value's key
+        default_value: This argument is returned if value is not found in database
+
+    Returns:
+        Any: value stored in DB
+
+    Raises:
+        ValueError: Raised if value type change fails
+    """
+
+    db_value = StorageData.get(module.qualified_name, guild_id, key)
+    if not db_value:
+        return default_value
+
+    t = locate(db_value.type)
+
+    if not t:
+        return default_value
+
+    value = t(db_value.value)
+
+    return value
+
+
+def exists(module: commands.Cog, guild_id: int, key: str) -> bool:
+    """Checks if data for module, key and guild_id combination
+    are present in DB.
+
+    Args:
+        module (:class:`nextcord.ext.commands.Cog`): module connected with the value
+        guild_id (:class:'int'): ID of guild connected with the value (0 for global)
+        key (:class:'str'): Value's key
+
+        Returns:
+            True if value exists in DB, False otherwise
+    """
+
+    db_value = StorageData.get(module.qualified_name, guild_id, key)
+
+    return db_value is not None
+
+
+def get_type(module: commands.Cog, guild_id: int, key: str) -> type:
+    """Get data type.
```

Comment:
The same as above. We're already performing the database lookup, it's easier to have only one function to get the value and then call
```py
item = pie.storage.get(...)
if type(item) is ...:
```

Reply:
Can you elaborate on that? I can't imagne any reason for that. If the `pie.storage` is reliable (which it will be) there shouldn't be any need for that.

Reply:
For example if you used int and now you want to use float, you can check when loading module that your set value is correct type and if not, convert it to float.

In this case it's more efficient to just get type of stored value instead of taking value, changing it's type and then converting it to type again to check.